### ISSUE TITLE
Rust enum information lost when generating Typescript

### ISSUE THREAD

**Issue by SebastienGllmt** [dcSpark/oura-postgres-sink#21]
When defining our database schema, we made enums (such as the block era) be represented at the database level by plain ints. The problem with this is that later on when we use Prisma to generate the Typescript type definition, it means they end up being represented in Typescript as just plain `number` which is not typesafe and also hurts readability.

I'm not sure if Prisma supports SQL enums, but it's something to investigate

**Comment by Mercurial** [comment]
just commenting because I've used Prisma before, it does seem like Prisma supports enums depending on which db 
<img width="732" alt="image" src="https://user-images.githubusercontent.com/585569/162660126-37a218b1-5342-46d6-98b3-2c520d4b01a8.png">
https://www.prisma.io/docs/reference/database-reference/database-features#misc

**Comment by SebastienGllmt** [comment]
Yeah I'm really disappointed with Prisma. Looks good on paper, in practice it's missing every feature we need

**Comment by CyberCyclone** [comment]
@SebastienGllmt I'm curious to what features you're saying don't exist in Prisma and what alternative client you would use that does cover those features?

**Comment by SebastienGllmt** [comment]
@CyberCyclone I want typed raw SQL, not an ORM. This is provided by pgtyped
